internal/manifests/targetallocator/container.go

```diff
@@ -110,6 +111,24 @@ func Container(cfg config.Config, logger logr.Logger, instance v1alpha1.TargetAl
 		},
 	}
 
+	if cfg.CertManagerAvailability() == certmanager.Available {
+		ports = append(ports, corev1.ContainerPort{
+			Name:          "https",
+			ContainerPort: 8443,
+			Protocol:      corev1.ProtocolTCP,
+		})
+		volumeMounts = append(volumeMounts, corev1.VolumeMount{
+			Name:      naming.TAServerCertificate(instance.Name),
+			MountPath: "/tls",
+		})
+		args = append(args,
+			"--enable-https-server",
+			"--https-ca-file=/tls/ca.crt",
+			"--https-tls-cert-file=/tls/tls.crt",
+			"--https-tls-key-file=/tls/tls.key",
+		)
```

Comment:
I would prefer if we set this in the configuration file instead.

Reply:
In https://github.com/open-telemetry/opentelemetry-operator/blob/main/cmd/otel-allocator/config/config.go the cli flags and their defaults override the file config. This means the config file is completely ignored ATM.
A quick fix can be to switch the order of LoadFromCLI and LoadFromFile. 
This will mean that if there are contradicting configs between a flag and an entry in the config yaml the config yaml will take precedent.
Can you see any issues with that?

Reply:
That's just a bug imo, what should really happen there is that flag values should take precedence if the flags are actually set. I ran into the same problem in https://github.com/open-telemetry/opentelemetry-operator/pull/3024, you can have a look at my solution.